app/views/unattended/report_templates/jobs_-_invocation_report_template.erb

```diff
@@ -0,0 +1,43 @@
+<%#
+name: Jobs - Invocation report template
+snippet: false
+template_inputs:
+- name: job_id
+  required: true
+  input_type: user
+  description: ID of job to report
+  advanced: false
+  value_type: plain
+  resource_type: JobInvocation
+  hidden_value: false
+- name: hosts
+  required: false
+  input_type: user
+  description: Field for filter hosts of job invocation. Leave blank for all hosts.
+  advanced: false
+  value_type: search
+  resource_type: Host
+  hidden_value: false
+model: ReportTemplate
+require:
+- plugin: foreman_remote_execution
+￼ version: 4.4.0
+-%>
+<%- invocation = find_job_invocation_by_id(input('job_id')) -%>
+<%- input_hosts = input('hosts') -%>
+<%- input_hosts = '*' if input_hosts.blank? -%>
+<%- load_hosts(search: "#{input_hosts} and job_invocation.id = #{input('job_id')}").each do |batch| -%>
```

Comment:
Could we rework this a bit? While searching for `* and job_invocation.id = $id` technically works, doing a wildcard search over all host fields is a bit excessive. Something along the lines of this

```suggestion
<%- parts = ["job_invocation.id = #{input('job_id')}"] %>
<%- parts << input('hosts') unless input('hosts').blank? %>
<%- search = parts.map { |part| '(' + part + ')' }.join(' AND ') %>
<%- load_hosts(search: search).each do |batch| -%>
```

Reply:
Thanks for great idea!